File: src/csharp/pulse/cdm/patient/actions/SEHemothorax.cs
/* Distributed under the Apache License, Version 2.0.
   See accompanying NOTICE file for details.*/

namespace Pulse.CDM
{
  public class SEHemothorax : SEPatientAction
  {
    protected eSide side;
    protected SEScalar0To1 severity;
    protected SEScalarVolumePerTime flow_rate;

    public SEHemothorax()
    {
      side = eSide.NullSide;
      severity = null;
      flow_rate = null;
    }

    public override void Clear()
    {
      base.Clear();
      side = eSide.NullSide;
      if (severity != null)
        severity.Invalidate();
      if (flow_rate != null)
        flow_rate.Invalidate();
    }

    public override bool IsValid()
    {
      return HasSide() && (HasFlowRate() || HasSeverity());
    }

    public eSide GetSide()
    {
      return side;
    }
    public void SetSide(eSide side)
    {
      this.side = side;
    }
    public bool HasSide()
    {
      return side == eSide.NullSide ? false : true;
    }

    public bool HasSeverity()
    {
      return severity == null ? false : severity.IsValid();
    }
    public SEScalar0To1 GetSeverity()
    {
      if (severity == null)
        severity = new SEScalar0To1();
      return severity;
    }

    public bool HasFlowRate()
    {
      return flow_rate == null ? false : flow_rate.IsValid();
    }
    public SEScalarVolumePerTime GetFlowRate()
    {
      if (flow_rate == null)
        flow_rate = new SEScalarVolumePerTime();
      return flow_rate;
    }

  }
}

File: src/java/com/kitware/pulse/cdm/patient/actions/SEHemothorax.java
/* Distributed under the Apache License, Version 2.0.
   See accompanying NOTICE file for details.*/

package com.kitware.pulse.cdm.patient.actions;

import com.kitware.pulse.cdm.bind.Enums.eSide;
import com.kitware.pulse.cdm.bind.PatientActions.HemothoraxData;
import com.kitware.pulse.cdm.properties.SEScalar0To1;
import com.kitware.pulse.cdm.properties.SEScalarVolumePerTime;
import com.kitware.pulse.cdm.properties.SEScalarVolume;

public class SEHemothorax extends SEPatientAction
{
  protected eSide side;
  protected SEScalar0To1 severity;
  protected SEScalarVolumePerTime flowRate;

  public SEHemothorax()
  {
    side = eSide.NullSide;
    severity = null;
    flowRate = null;
  }

  public void copy(SEHemothorax other)
  {
    if(this==other)
      return;
    super.copy(other);
    side = other.side;
    if (other.severity != null)
      getSeverity().set(other.severity);
    else if (severity != null)
      severity.invalidate();
    if (other.flowRate != null)
      getFlowRate().set(other.getFlowRate());
    else if (flowRate != null)
      flowRate.invalidate();
  }

  @Override
  public void clear()
  {
    super.clear();
    side = eSide.NullSide;
    if (severity != null)
      severity.invalidate();
    if (flowRate != null)
      flowRate.invalidate();
  }

  @Override
  public boolean isValid()
  {
    return hasSide() && (hasFlowRate() || hasSeverity());
  }

  public static void load(HemothoraxData src, SEHemothorax dst)
  {
    SEPatientAction.load(src.getPatientAction(), dst);
    if(src.getSide()!=eSide.UNRECOGNIZED)
    	dst.side = src.getSide();
    if(src.hasSeverity())
      SEScalar0To1.load(src.getSeverity(),dst.getSeverity());
    if(src.hasFlowRate())
      SEScalarVolumePerTime.load(src.getFlowRate(),dst.getFlowRate());
  }

  public static HemothoraxData unload(SEHemothorax src)
  {
    HemothoraxData.Builder dst = HemothoraxData.newBuilder();
    unload(src,dst);
    return dst.build();
  }

  protected static void unload(SEHemothorax src, HemothoraxData.Builder dst)
  {
    SEPatientAction.unload(src,dst.getPatientActionBuilder());
    if (src.hasSide())
      dst.setSide(src.side);
    if (src.hasSeverity())
      dst.setSeverity(SEScalar0To1.unload(src.severity));
    if (src.hasFlowRate())
      dst.setFlowRate(SEScalarVolumePerTime.unload(src.flowRate));
  }

  public eSide getSide()
  {
    return side;
  }
  public void setSide(eSide side)
  {
    this.side = side;
  }
  public boolean hasSide()
  {
    return side == eSide.NullSide ? false : true;
  }

  public boolean hasSeverity()
  {
    return severity == null ? false : severity.isValid();
  }
  public SEScalar0To1 getSeverity()
  {
    if (severity == null)
      severity = new SEScalar0To1();
    return severity;
  }

  public boolean hasFlowRate()
  {
    return flowRate == null ? false : flowRate.isValid();
  }
  public SEScalarVolumePerTime getFlowRate()
  {
    if (flowRate == null)
      flowRate = new SEScalarVolumePerTime();
    return flowRate;
  }

  @Override
  public String toString()
  {
    return "Hemothorax"
        + "\n\tSide : " + getSide()
        + "\n\tSeverity: " + getSeverity()
        + "\n\tFlowRate: " + getFlowRate();

  }

}
